syntax = "proto3";

message DamagerItem {
  uint64 ObjectGuid  = 1;
  int32	ObjStatue  = 2;		//实例状态
  float x = 3;
  float y = 4;
  float z = 5;
  int32 chghp = 6;  
}

message ActionItem {
	uint64  ObjectGuid = 1;
	int32	ObjStatue  = 2;		//实例状态
	int32	ActorID    = 3;		//类型ID
	uint64  ControlerID = 4;    //AI控制人的GUID
	int32   ActionID = 5;
	int32   ActionState = 6; 
	float   x = 7;
	float   y = 8;
	float   z = 9;
	float   ft = 10;
	int32   hp = 11;
	int32   mp = 12;
	int32   hpmax = 13;
	int32   mpmax = 14;
	repeated DamagerItem DamagerList = 15; 
}

message ObjectActionReq {
   repeated ActionItem ActionList = 1;
}

message ObjectActionNty {
   repeated ActionItem ActionList = 1;
}

message NewItem {
	uint64 ObjectGuid = 1;		//实例ID
	int32  ObjType    = 2;		//类别
	int32 ObjStatue  = 3;		//实例状态
	int32  ActorID    = 4;		//类型ID
	string Name       = 5;
	int32  Level      = 6;
	uint64 HostGuid   = 7;		//主人的GUID
    uint64 ControlerID = 8;     //AI控制人的GUID
	uint64 SummonID = 9;		//招唤者的GUID
	float  x = 10;
	float  y = 11;
	float  z = 12;
	float  ft = 13;
	int32  hp = 14;
	int32  mp = 15;
	int32  hpmax = 16;
	int32  mpmax = 17;
	int32  weapon = 18; //武器
}

message ObjectNewNty {
  repeated NewItem NewList = 1;
}

message ObjectRemoveNty {
  repeated uint64 RemoveList = 1;
}

message HeartBeatReq {
  uint32  TimeStamp = 1;
}

message HeartBeatAck {
  uint32  TimeStamp = 1;
  uint32  ServerTime = 2;
}


message ObjectDieNotify {
  uint64  ObjectGuid = 1;
}